runtime/src/payout.rs

```diff
@@ -0,0 +1,208 @@
+        (1f64 / ExpPrecision::get()) as u64,
+    )
+}
+
+/// Implements a custom inflation model based on the following formula:
+/// I(s_c) = I_b + I_v(s_c)
+///
+/// with:
+/// I_v(x) = C * exp(K * (1 - (s_c / s_t)))
+///
+/// where:
+/// I_b: base inflation
+/// I_v: variable inflation
+/// s_t: target inflation
+/// K: sensitivity coefficient
+/// C: multiplier
+pub struct ZKVPayout;
+
+parameter_types! {
+    /// The target precions for exp(); impacts on the final precision for inflation computation.
+    pub const ExpPrecision: f64 = 10e-15f64;
+    /// Base inflation (I_b).
+    pub InflationBase: Perquintill = Perquintill::from_percent(5);
+    /// The optimal staking rate (s_t).
+    pub StakingTarget: Percent = Percent::from_percent(50);
+    /// Sensitivity coefficient (K).
+    pub const K: i128 = 1i128;
+    /// Multiplier (C).
+    pub const C: u128 = 1u128;
+    /// Percentage of the minted tokens that goes to the validators.
+    pub ValidatorsSplit: Percent = Percent::from_percent(100);
+    /// Percentage of the minted tokens that goes to the rest (i.e. treasury).
+    pub OthersSplit: Percent = Percent::from_percent(0);
+}
+
+impl pallet_staking::EraPayout<Balance> for ZKVPayout {
+    /// Calculates the validators reward based on the duration of the era.
+    fn era_payout(
+        total_staked: Balance,
+        total_issuance: Balance,
+        era_duration_millis: u64,
+    ) -> (Balance, Balance) {
+        if total_issuance == 0 {
+            return (0, 0);
+        }
+
+        const MILLISECS_PER_YEAR: u64 = 1000 * 60 * 60 * 24 * (36525 / 100);
+        let time_portion = Perquintill::from_rational(era_duration_millis, MILLISECS_PER_YEAR);
+
+        let staking_current: Perbill = Perbill::from_rational(total_staked, total_issuance);
+        let s = StakingTarget::get()
+            .saturating_reciprocal_mul(staking_current * (1f64 / ExpPrecision::get()) as u128);
+        let exp_arg = ((1f64 / ExpPrecision::get()) as i128 - s as i128) * K::get();
+        let inflation_var = to_inflation(exp_arg as f64 / (1f64 / ExpPrecision::get()));
+
+        let inflation_tot: Balance =
+            C::get() * (time_portion * (InflationBase::get() + inflation_var) * total_issuance);
+
+        (
+            ValidatorsSplit::get() * inflation_tot,
+            OthersSplit::get() * inflation_tot,
+        )
+    }
+}
+
+/// This module tests the correct computation of rewards for validators.
+#[cfg(test)]
+use pallet_staking::EraPayout;
+
+#[cfg(test)]
+use crate::{Balance, Perbill, Runtime, ACME, MILLICENTS};
+
+#[test]
+fn check_params_sanity() {
+    // staking_target should be at least 0.5 (and so 0 <= s_c / s_t <= 2)
+    assert!(StakingTarget::get() * 10u16 >= 5u16);
+    // base inflation is 5%
+    assert!(InflationBase::get() * 100u64 == 5u64);
+    // The sum of the splits is exactly 100%
+    assert!(ValidatorsSplit::get() * 100u16 + OthersSplit::get() * 100u16 == 100u16);
+}
```

Comment:
Do we have some reference about why these numbers are _correct_ and what are the valid range in any case?

The second assert seams to say any other number is just wrong. Maybe can be enough to be explicit in the assert message to alert the developer where to look to check if the new value is correct.

Reply:
:+1: